gradle-consistent-versions-idea-plugin/src/main/java/com/palantir/gradle/versions/intellij/RepositoryExplorer.java

```diff
@@ -104,21 +113,49 @@ private Set<GroupPartOrPackageName> fetchFoldersFromContent(String contents) {
     }
 
     private Set<DependencyVersion> parseVersionsFromContent(String content) {
-        Set<DependencyVersion> versions = new LinkedHashSet<>();
         try {
             XmlMapper xmlMapper = new XmlMapper();
 
             Metadata metadata = xmlMapper.readValue(content, Metadata.class);
-            if (metadata.versioning() != null && metadata.versioning().versions() != null) {
-                String latest = metadata.versioning().latest();
-                for (String version : metadata.versioning().versions()) {
-                    versions.add(DependencyVersion.of(version, latest.equals(version)));
-                }
-            }
+            return parseVersionsFromContent(metadata);
         } catch (Exception e) {
             log.error("Failed to parse maven-metadata.xml", e);
         }
-        return versions;
+        return Collections.emptySet();
+    }
+
+    @VisibleForTesting
+    final Set<DependencyVersion> parseVersionsFromContent(Metadata metadata) {
+        List<String> allVersions = new ArrayList<>(metadata.versioning().versions());
+        Collections.reverse(allVersions);
+
+        if (allVersions.isEmpty()) {
+            return Collections.emptySet();
+        }
+
+        String releaseOrLatestVersion = Optional.ofNullable(
+                        metadata.versioning().release())
+                .filter(l -> !l.isEmpty())
+                .orElseGet(() -> metadata.versioning().latest());
+
+        // Check if the releaseOrLatestVersion is stable, it not find first stable version, if no stable versions return
+        // the releaseOrLatestVersion
+        String latestStableVersion = Optional.of(releaseOrLatestVersion)
+                .filter(this::isStableVersion)
+                .or(() -> allVersions.stream().filter(this::isStableVersion).findFirst())
+                .orElse(releaseOrLatestVersion);
+
+        Collections.reverse(allVersions);
```

Comment:
Rather than reversing the `allVersions` list twice, can we just avoid reversing it mutably and use `Lists.reverse` where needed? This make a list "view" where the list is reversed without actually changing it.